docs/autoprocess.py

```diff
@@ -0,0 +1,412 @@
+                    return filtered_dict
+
+            else:
+                raise ValueError("Invalid argument given to autoprocess directive.")
+
+    def get_all_processes(self, process_dir, github_base_url):
+        """
+        Return a dictionary of all processes in `process_dir`
+
+        Dictionary key is an URL's to process source code on resolwe-bio
+        GitHub page and value is process as it is loaded from yaml file.
+
+        :param process_dir: Parent directory of all process files
+        :type process_dir: path/str
+        :param github_base_url: Base of the url to resolwe-bio on GitHub
+        :type github_base_url: url/string
+
+        :rtype: dict
+        """
+        all_process_files = []
+        process_file_extensions = ['*.yaml', '*.yml']
+        for root, _, filenames in os.walk(process_dir):
+            for extension in process_file_extensions:
+                for filename in fnmatch.filter(filenames, extension):
+                    all_process_files.append(os.path.join(root, filename))
+
+        def read_yaml_file(fname):
+            with open(fname) as file_:
+                return yaml.load(file_)
+
+        all_processes = {}
+        for yaml_file in all_process_files:
+            processes_in_file = read_yaml_file(yaml_file)
+            for process in processes_in_file:
+
+                # This section finds the line in file where the
+                # defintion of the process starts. (there are
+                # multiple process definition in some files).
+                i = 1
+                found = False
+                with open(yaml_file) as file_:
+                    for line in file_:
+                        if 'slug: ' + process['slug'] in line:
+                            found = True
+                            break
+                        i += 1
+                    if not found:
+                        # In case starting line is not found just
+                        # return first line of the file
+                        i = 1
+                startline = i
+
+                # Put together URL to process definition on GitHub:
+                cutted_yaml_path = yaml_file.split('/resolwe_bio/processes/')
+                github_url = github_base_url + cutted_yaml_path[1] + '#L' + str(startline)
+                chosen_process = [proc for proc in processes_in_file if proc['slug'] == process['slug']]
+                assert len(chosen_process) == 1
+
+                all_processes[github_url] = chosen_process[0]
+
+        return all_processes
+
+    def get_all_categories(self, process_list):
+        """
+        Get list of all categories in `process_list`.
+
+        For processes with undefined category field (yes, there are some!),
+        there is uncategorized category.
+
+        :param process_list: list of all processes
+        :type process_list: list
+
+        :rtype: list
+        """
+        categories = ['uncategorized']
```

Comment:
categories = set()
    for process in process_list:
        categories.add(process.get('category', 'uncategorized'))
    return list(categories)